wasm/src/record/record_plaintext.rs

```diff
@@ -43,6 +46,16 @@ impl RecordPlaintext {
     pub fn gates(&self) -> u64 {
         ***self.0.gates()
     }
+
+    #[wasm_bindgen(js_name = serialNumberString)]
+    pub fn serial_number_string(&self, private_key: &PrivateKey, program_id: &str, record_name: &str) -> String {
+        let parsed_program_id = ProgramIDNative::from_str(program_id).unwrap();
+        let record_identifier = IdentiferNative::from_str(record_name).unwrap();
+        let commitment = self.to_commitment(&parsed_program_id, &record_identifier).unwrap();
+
+        let serial_number = RecordPlaintextNative::serial_number(private_key.clone().into(), commitment).unwrap();
+        serial_number.to_string()
+    }
```

Comment:
One final nit, we generally want to avoid `unwrap()` within Rust in general when it's possible the code will error. In the `wasm` context, unwrapping will cause a panic in the `wasm` environment which cannot be handled in the javascript environment other than a try/catch statement. Even if you do this in javascript the errors are indecipherable in a general wasm panic. Creating a Result Enum to handle errors allows the developer to debug their code better.

Specifically the `Result` types in SnarkVM are of type `anyhow::Result` which is a `Result` type from a crate called `anyhow`. Wasm doesn't directly support this so we use an error transformation covnenience method to transform this type to a Result<String, String> which the wasm ffi supports.

Also secondly, any type that implements the `Copy` trait does not need to be cloned as rust takes care of copying the memory directly. So you can pass by value. Or in the case that you have a reference like this function you've defined you can copy by using the `deref` operator.

Thus you have
```rust
#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
pub struct PrivateKey<N: Network> {
```

And you can `deref` like so
```rust
*private_key
// Or in the case you're using it
*private_key.into()
```


```suggestion
    pub fn serial_number_string(&self, private_key: &PrivateKey, program_id: &str, record_name: &str) -> Result<String, String> {
        let parsed_program_id = ProgramIDNative::from_str(program_id).map_err(|_| "Invalid ProgramID specified".to_string());
        let record_identifier = IdentiferNative::from_str(record_name).map_err(|_| "Invalid Identifier specified for record".to_string());
        let commitment = self.to_commitment(&parsed_program_id, &record_identifier).map_err(|_| "Record does not have a commitment and is likely either invalid or unspent".to_string())

        let serial_number = RecordPlaintextNative::serial_number(*private_key.into(), commitment).map_err(|_| "Serial number derivation failed".to_string());
        serial_number.to_string()
    }
```

Reply:
Thanks for the explanation of why we want to use `Result` instead of just `unwrap`; mapping the errors like you've outlined will be a much better experience for javascript clients. The copy instead of clone suggestion makes sense to me too. Only thing I would change is the error message on the `to_commitment` function -- since this is just a hash of the program id, the record name, and the data of the record itself, this should only error if hashing or converting the plaintext record object to bits fails. We should be able to compute the commitment even if the record is unspent. So maybe a better message is `"A commitment for this record and program could not be computed, likely because the record data is invalid"`?